src/tests/test_module.py

```diff
@@ -66,7 +66,8 @@ def testModuleInterface(self):
         self.assertEquals(type(System.__dict__), type({}))
         self.assertEquals(System.__name__, 'System')
         # the filename can be any module from the System namespace (eg System.Data.dll or System.dll)
-        self.assertTrue(fnmatch(System.__file__, "*System*.dll"))
+        self.assertTrue(fnmatch(System.__file__, "*System*.dll"),
+                        "unexpected System.__file__" + System.__file__)
```

Comment:
The System namespace is spread out across a couple of DLLs: mscorlib.dll, System.dll, and all the other System.*.dlls. In ModuleObject we iterate through all the known assemblies for a namespace and pick the last one for the filename. The iteration order would be the on the hash of the assembly, so it should have always been unpredictable whether you get mscorlib.dll or one of the others.
Let me change the test such that mscorlib.dll is also accepted.